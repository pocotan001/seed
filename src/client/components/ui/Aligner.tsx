import { CSSProperties } from "react";
import styled from "styled-components";
import margin, { MarginProps } from "../styles/extends/margin";
import { px } from "../styles/utils";

interface OwnProps {
  children: React.ReactNode;
  direction?: CSSProperties["flexDirection"];
  justify?: CSSProperties["justifyContent"];
  align?: CSSProperties["alignItems"];
  width?: CSSProperties["width"];
  height?: CSSProperties["height"];
}

type AlignerProps = OwnProps & MarginProps;

const Aligner = styled("div")<AlignerProps>`
  display: flex;
  flex-wrap: wrap;
  ${({ direction }) => direction && `flex-direction: ${direction}`};
  ${({ justify }) => justify && `justify-content: ${justify}`};
  ${({ align }) => align && `align-items: ${align}`};
  ${({ width }) => width && `width: ${px(width)}`};
  ${({ height }) => height && `height: ${px(height)}`};
  ${margin};
`;

export default Aligner;
