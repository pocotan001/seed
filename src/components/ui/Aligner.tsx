import { CSSProperties } from "react";
import styled, { css } from "styled-components";
import withMargin, {
  MarginProps
} from "~/components/styles/extends/withMargin";
import withMedia, {
  MediaProps,
  StyleFactory
} from "~/components/styles/extends/withMedia";
import { px } from "~/utils";

interface AlignerProps
  extends AlignerStyleProps,
    MediaProps<AlignerStyleProps>,
    MarginProps {
  children: React.ReactNode;
}

interface AlignerStyleProps {
  direction?: CSSProperties["flexDirection"];
  justify?: CSSProperties["justifyContent"];
  align?: CSSProperties["alignItems"];
  width?: CSSProperties["width"];
  height?: CSSProperties["height"];
}

const styles: StyleFactory<AlignerStyleProps> = ({
  direction,
  justify,
  align,
  width,
  height
}) => css`
  ${direction && `flex-direction: ${direction}`};
  ${justify && `justify-content: ${justify}`};
  ${align && `align-items: ${align}`};
  ${width && `width: ${px(width)}`};
  ${height && `height: ${px(height)}`};
`;

const Aligner = styled<AlignerProps, "div">("div")`
  display: flex;
  flex-wrap: wrap;
  ${styles};
  ${withMedia(styles)};
  ${withMargin};
`;

export default Aligner;
