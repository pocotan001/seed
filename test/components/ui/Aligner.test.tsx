import * as React from "react";
import * as renderer from "react-test-renderer";
import Aligner from "~/components/ui/Aligner";

describe("<Aligner>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Aligner>alo</Aligner>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
