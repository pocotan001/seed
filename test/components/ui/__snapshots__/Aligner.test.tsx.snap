// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aligner> should render correctly 1`] = `
<div
  className="sc-bdVaJa ijCOvj"
>
  alo
</div>
`;
